services/cd-service/pkg/repository/queue.go

```diff
@@ -54,12 +55,23 @@ func (q *queue) add(ctx context.Context, transformers []Transformer) <-chan erro
 		transformers: transformers,
 		result:       resultChannel,
 	}
+
 	select {
 	case q.transformerBatches <- e:
 		return resultChannel
 	case <-ctx.Done():
 		e.finish(ctx.Err())
 		return resultChannel
+	default:
+		//Channel is full, we don't want to put anything else there.
+		e.finish(&TransformerBatchApplyError{
```

Comment:
I think you can have this much simpler here.
Create an error on top-level like this:

```go
var ErrQueueFull error = fmt.Errorf("queue full")
```

Then you can use `e.finish(ErrQueueFull)` here and later when checking [here](https://github.com/freiheit-com/kuberpult/pull/1466/files#diff-1a2c7d4b2d78782768dc4449dbb8e6fb964eaf7a97d631faebd14adaec1fb6d9L322) you can use this:

```
if errors.Is(err, ErrQueueFull) {
   return nil, status.Error(codes.ResourceExhausted, "Could not process ProcessBatch request. Too many pending operations. Queue is full.")
}
```